### DISCUSSION TITLE
Versatz der Uhrzeit

### DISCUSSION THREAD

**Discussion by kingherold** [beabel/radiowecker#18] [Zeigen und berichten / Show and tell]
die Uhrzeit wird alle 60 Sekunden aktualisiert, d.h. der Minutenwechsel findet jede Minute mit gleichem Versatz zur Internetzeit statt, abhängig, wann der ESP gebootet wurde.

Ich habe Zeile 235 bei mir so abgeändert:
tick = millis() - ti.tm_sec * 1000;
// war so: tick = millis();

damit wird die aktuelle Sekunde mit eingerechnet und der nächste Minutenwechsel findet entsprechend eher statt. Damit korrigiert sich der Minutenwechsel nach der zweiten Minute Laufzeit.
Auf die Sekunde genau ist es aber auch noch nicht, weil die Abfrage der Internetzeit und die interne Verarbeitung ein bisschen dauern, hier könnte man zusätzlich noch einen individuellen Offset reinrechnen, falls es jemand ganz genau nimmt.

**Comment by beabel** [comment]
In develop übernommen.
Beim nächsten Release mit drin.
